apps/crm/src/modules/mailchimp/views/mailchimp-config-view/mailchimp-config-view.tsx

```diff
@@ -0,0 +1,75 @@
+import { trpcClient } from "../../../trpc/trpc-client";
+import { Box, Text } from "@saleor/macaw-ui/next";
+import { RemoveMailchimpConfig } from "../../config-sections/remove-mailchimp-config/remove-mailchimp-config";
+import { MailchimpAuthorizeView } from "../mailchimp-authorize-view/mailchimp-authorize-view";
+import { MailchimpLists } from "../../config-sections/mailchimp-lists/mailchimp-lists";
+import { WebhookConfiguration } from "../../config-sections/webhook-configuration/webhook-configuration";
+import { SaleorCustomersSync } from "../../../saleor-customers-sync/saleor-customers-sync";
+import { ConnectionStatus } from "../../config-sections/connection-status/connection-status";
+import { Instructions } from "../../config-sections/instructions/instructions";
+
+const header = (
+  <Box marginBottom={12}>
+    <Text as="h1" variant="title" size="large">
+      Mailchimp
+    </Text>
+  </Box>
+);
+
+export const MailchimpConfigView = () => {
+  const {
+    data: mailchimpConfigured,
+    refetch,
+    isFetched,
+    isLoading,
+    error,
+  } = trpcClient.mailchimp.config.getMailchimpConfigured.useQuery();
+
+  const isMailchimpConfigured = mailchimpConfigured?.configured;
+
+  if (isLoading) {
+    return (
+      <div>
+        {header}
+        <Text>Checking Mailchimp config status...</Text>
+      </div>
+    );
+  }
+
+  if (error) {
+    return (
+      <div>
+        {header}
+        <Box marginBottom={12} display="flex" justifyContent="space-between" gap={12}>
+          <ConnectionStatus status="error" __flex="0 1 50%" />
+        </Box>
+      </div>
+    );
+  }
+
+  if (isMailchimpConfigured && isFetched) {
+    return (
+      <div>
+        {header}
+        <Instructions marginBottom={12} />
+        <Box marginBottom={12} display="flex" justifyContent="space-between" gap={12}>
+          <ConnectionStatus status="ok" __flex="0 1 50%" />
+          <MailchimpLists __flex="0 1 50%" />
```

Comment:
Is there a reason for why the style prop `__flex` is passed here and not declared in the component itself?

Reply:
Yes, component should not be aware of its placement/parent. It must be reusable by the definition, so it should naturally fill its container. Here, at parent level, component is set to 50% width. Alternatively it can be placed inside a div/Box that will do that, but its less elegant - requires additional markup